Frontpage Tabs + Glue

ChrisLaganiere opened the pull request:
Integrating DJ List and Events pages into the new frontpage, using `react-router` for a single-page app. Some of the data used is still hardcoded.. Hopefully I will get the chance to use `redux` to set up data storage.

See the updated demo at [uclaradio.com/beta](https://uclaradio.com/beta) 🍉 

![screen shot 2016-12-15 at 9 23 02 pm](https://cloud.githubusercontent.com/assets/2042745/21252646/5b70d780-c30d-11e6-82f2-dc67ce97cd9d.png)

![screen shot 2016-12-15 at 9 24 03 pm](https://cloud.githubusercontent.com/assets/2042745/21252648/5f3f4c5c-c30d-11e6-8496-b2ef82851e3f.png)

hsuregan commented:
Hey guys looks awesome but can we think of a way to align the circles of the DJs? Can we think of alternative placements for the DJ names?  @binerys @ChrisLaganiere

ChrisLaganiere commented:
awesome point @hsuregan here's an issue for that: #52 

what we can do then is check out this `tuneup` branch, make changes, and create a new branch from that

kianadk commented:
Looks awesome! 😄 Do we want the events content to be centered? It was left-aligned in the designs and when I was working on it, but this looks good too 👍🏼
<img width="1280" alt="screen shot 2016-12-15 at 12 13 29 pm" src="https://cloud.githubusercontent.com/assets/10238224/21279616/f524801a-c395-11e6-8672-0718ba50850c.png">

kianadk commented:
Ooh also, the updated demo isn't working for me...I'm having to reload the whole page when I switch between tabs to see the new content

ChrisLaganiere commented:
Yeah! I'm sorry, it was totally working last night.. One of the changes I made was enabling [es6](http://es6-features.org), which broke the now playing sidebar.. and then it breaks when a show is active like this morning 😛 fixed now.

Umm let's get one alignment (left / middle) between all the pages? I like left aligned but it looks different on mobile

kianadk commented:
No worries! And I think I prefer left-aligned as well...I'll see what I can do with changing that for larger screens and keeping mobile as is

hsuregan commented on app.js:
```diff
@@ -63,12 +63,13 @@ app.use('/faq', function(req, res, next) {
 });
 
-// catch 404 and forward to error handler
+// catch 404 and forward to frontpage
```
what's happening here? @ChrisLaganiere

hsuregan commented on react/frontpage/reducers/index.js:
```diff
@@ -0,0 +1,14 @@
+// import { combineReducers } from 'redux'
```
what does this do? @ChrisLaganiere

ChrisLaganiere replied:
saving for later

ChrisLaganiere commented:
Hey I made some changes over the weekend! 🙂 The folder looks a bit different now

![screen shot 2016-12-18 at 6 36 52 pm](https://cloud.githubusercontent.com/assets/2042745/21299486/026079a2-c551-11e6-8a4b-b284c8383922.png)


* Switched us to use [es6](http://es6-features.org/) for webpack

so you can use arrow functions, etc. in the compiled react files

* Switched us to use [Sass](http://sass-lang.com) for webpack -> converted everything from `/public/build/css/*.css` to `/react/*.scss`
  * every component except `DJList`, left those styles in `radio.css`

So now you can split up styling by react components, just use `require('./Component.scss')` in each component 🌵

* Moved stateful logic to a redux store... So now we use reducers and action creators to update data, all the tabs are set up with that now. All the data is coming from our api

^ Redux is super cool! The codebase is reconfigured a bit with new folders, make sure to check that out. The result is that all the fetched data is collected, and we can move all the data logic out of the presentation components.

* Added a promo banner, how does that look?

Keep it up! 🍭

hsuregan commented:
Looks great :)

Currently adding the chat feature is there anything I need to watch out for on the frontend? Branched off of master without es6 or sass

hsuregan commented:
@ChrisLaganiere

nmockovciak commented:
Thoughts:
• I love the decision of getting rid of the info box and just putting the logo above the show blurb! Maybe we can even get rid of the "UCLA Radio is an entirely student-run radio station..." text and just devote an entire tab to info/contact like the one I made a mockup for? 
• Also I'm not sure if the logo would look better black (using the png below)? 

![ucla-radio-logo-2_720](https://cloud.githubusercontent.com/assets/10100525/21333273/7f344136-c603-11e6-9fc0-a6be60b2cd61.png)

• Still not crazy about the **banner** though tbh-- did anyone have thoughts about that? I feel like the dimensions it has now aren't bad but you can't see the whole thing and making it any bigger would take up too much space? Have we considered instead of having a banner, just making it look like a normal post and incorporating it into the waterfall content and fixing it at the top (if that's possible)? Or even putting the banner below the tabs could be better maybe? I could never really figure out how to make the banner look good either lol so maybe we should get rid of them altogether, if other departments wouldn't be too angry? 

• Maybe, we could replace the "Now Playing" with "Current Show" or something like that for more clarity?

• For the "**More**" tab, is it possible to make the additional tabs drop down instead of the current functionality? It makes it look even more cluttered on mobile :/

• **Events** tab looks so good @kianadk ! The only problem seems to be that the text keeps getting cut off for both the band name and the venue name? Otherwise, great work 

• Lastly, same sentiments as everyone else about spacing for the **DJs** tab, but it looks pretty good too! Maybe to keep consistency, we could play around with putting the DJ names on top of the circles and letting the names disappear when you hover over them (sort of like the events)? But that might get tricky, and it looks ok as is, just a suggestion

Thanks everyone for all the hard work. I feel bad for not contributing to this stage much so far, but I'm ready to dive in and learn what I can between now and the launch :-)

ChrisLaganiere commented:
Sounds great I will address those @nmockovciak :)

@hsuregan yep that's just fine, can't wait to see it!

hsuregan commented:
lmao this will definitely break my commit 😭 ... no worries tho... ~ jk .... ~ jk again 
*hysterical* *not looking forward to merge conflicts*

ChrisLaganiere commented:
Don't worry @hsuregan ! haha I can do merging 😛

hsuregan commented:
@ChrisLaganiere ugh it's ok I'll just bite the bullet.  You can go ahead with merging this branch into master I'll submit a pull request agains master as soon as I'm done

ChrisLaganiere commented:
Merging this in now.. Good job everybody! :)
